engine/step/step.go

```diff
@@ -521,6 +538,23 @@ func AfterRun(st *Step, values *values.Values, ss StateSetter) {
 		return
 	}
 
+	// Check if we have a function as runner or not.
+	runner, err := getRunner(st.Action.Type)
+	if err != nil {
+		ss(st.Name, StateServerError, err.Error())
+		return
+	}
+
+	_, isFunction := runner.(*functions.Function)
+	if isFunction {
+		var functionInput map[string]interface{}
+		if err := utils.JSONnumberUnmarshal(bytes.NewBuffer(st.Action.Configuration), &functionInput); err != nil {
+			ss(st.Name, StateServerError, err.Error())
+			return
+		}
+		values.SetFunctionsArgs(functionInput)
```

Comment:
Here, we previously used `values.Clone` to prevent alter on the `values` object of the caller.
But, in this function, we have a second parameter, which is `ss` for `StateSetter`: https://github.com/ovh/utask/blob/master/engine/step/step.go#L521 
Called from here: https://github.com/ovh/utask/blob/master/engine/engine.go#L431
definition of this function is here: https://github.com/ovh/utask/blob/master/engine/engine.go#L1064-L1071

Everytime a condition is matching, we need to alter the `state` of some steps, using `ss` (such as here https://github.com/ovh/utask/blob/master/engine/step/step.go#L551 )
And the call of the closure in `resolutionStateSetter` is using the parent `values` object to alter the current state.
So when the second condition is evaluated, and require that the first condition altered the state to match, we are using the cloned value, which has not been altered by the closure in `resolutionStateSetter`, which mean that the condition state will not match, making the test failing.

I guess not cloning the values is fine, but the drawback of function_args not working in conditions if they are nested seems a huge drawback that we should address before merging this one